docs/windows_dev.md

```diff
@@ -0,0 +1,99 @@
+    (dlv) c
+    time="2024-07-01T14:46:21+03:00" level=info msg="OCI Worker not supported on Windows."
+    > [Breakpoint 1] github.com/moby/buildkit/frontend/dockerfile.testEnvEmptyFormatting() C:/dev/core-containers/buildkit/frontend/dockerfile/dockerfile_test.go:334 (hits goroutine(51):1 total:1) (PC: 0x1aecfd6)
+    329:                         require.Equal(t, x.expected, string(dt))
+    330:                 })
+    331:         }
+    332: }
+    333:
+    => 334: func testEnvEmptyFormatting(t *testing.T, sb integration.Sandbox) {
+    335:         f := getFrontend(t, sb)
+    336:
+    337:         dockerfile := []byte(integration.SelectTestCase(t, []string{
+    338:                 `
+    339: FROM busybox AS build
+    (dlv)
+```
+#### Visual Studio Code
+For debugging the tests in vscode copy the following configuration into your launch.json file and run the debugger.
+
+```
+{    
+    "version": "0.2.0",
+    "configurations": [ {
+        "name": "Test buildkit tests",
+        "type": "go",
+        "request": "launch",
+        "mode": "test",
+        "program": "${workspaceFolder}\\frontend\\dockerfile\\dockerfile_test.go",
+        "showLog": true,
+        "args":["-test.run=NameOfPackage/NameOfTestMethod/worker=containerd"]
+    } ]
+}
+```
+#### Goland
+Copy the following configurations to the Run/Debug configurations pop-up window. Each of the following items consists of the label and input for that label - leave everything else as blank or with its default value. 
+
+```
+Test framework    => gotest
+Test kind         => Package
+Package path      => github.com/moby/buildkit/frontend/dockerfile
+Pattern           => ^\QTestIntegration\E$ 
+Working directory => C:\buildkit\frontend\dockerfile (this is the full path of the package)
+Program arguments => -test.run=^TestIntegration/NameOfTestMethod/.*
+```
+
+> Name of the method being tested **starts with a capital letter** for both vscode and goland. Replace _NameOfTestMethod_ with the specific method you're testing.
+
+### Running the whole test-suite
+Running the whole testsuite can take a long time, and it's not advisable to do it on your local machine. However, if you'd like to run it, you can do so with the following command:
+
+> Make sure to install gotestsum => go install gotest.tools/gotestsum@latest
+ ```
+gotestsum `
+--jsonfile="./bin/testreports/go-test-report-test-windows-amd64-frontenddockerfile--containerd.json" `
+--junitfile="./bin/testreports/junit-report-test-windows-amd64-frontenddockerfile--containerd.xml"
+--packages="./frontend/dockerfile" `
+-- "-mod=vendor" "-coverprofile" "./bin/testreports/coverage-test-windows-amd64-frontenddockerfile--containerd.txt"  "-covermode" "atomic" `
+-v --parallel=6 --timeout=60m `
+**--run=TestIntegration/.*/worker=containerd**
+```
+
+<!-- TBD: the exact pattern variants for --run to be enumerated and explained. -->
```

Comment:
This is a TBD for which section?

Reply:
This is a TBD for the arguments you can add after "run". I'm actually going to remove it because its beyond the scope of this document.